Before: 0fbd4630901ab5a4471edb7b151bcf0dbfc989c14f9aa190c6446afa6a7b069e
After: 33bfbc981a0607f4ba3019729ce32d3a0171520c560873cc5a84d42f841bc3e9
Patch: 01ad980ff95f556540dab2d6e216e30458f35e1b5d9cf306a5d864757fd37b45
--- %project_name%.gemspec.tt	2012-09-07 15:57:57.817373557 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,23 +0,0 @@
-# encoding: utf-8
-require File.expand_path('../lib/<%= project_name %>/version', __FILE__)
-
-Gem::Specification.new do |gem|
-  gem.name        = "<%= project_name %>"
-  gem.version     = <%= project_module %>::VERSION.dup
-  gem.date        = Time.new.strftime '%Y-%m-%d'
-  #gem.summary     = %q{TODO: Write a gem summary}
-  #gem.description = %q{TODO: Write a gem description}
-  <%= author_name_comment  %>gem.authors     = ["<%= author_name  || 'TODO' %>"]
-  <%= author_email_comment %>gem.email       = ["<%= author_email || 'TODO' %>"]
-  #gem.homepage    = "https://github.com/username/<%= project_name %>"
-
-  gem.files         = File.exists?('.git') ? `git ls-files`.split($\) : \
-                      Dir['**/*'].reject{|f| !File.file? f or %r!^(?:target|resources/latest)/! === f}.sort
-  gem.test_files    = gem.files.grep(/^test\//)
-  gem.require_paths = %w[lib]
-  gem.bindir        = 'bin'
-  gem.executables   = %w[]
-
-  gem.add_runtime_dependency 'corvid'
-end
-
--- Gemfile	2012-09-06 07:57:58.255746233 +1000
+++ Gemfile	2012-09-07 16:23:57.122016195 +1000
@@ -1,5 +1,4 @@
 source :rubygems
-gemspec
 
 # Load Corvid dependencies
 eval File.read(File.expand_path '../.corvid/Gemfile', __FILE__)
--- bin/%plugin_name%.tt	2012-09-06 07:57:58.255746233 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,21 +0,0 @@
-#!/usr/bin/env ruby
-# This file is generated by Corvid.
-
-# Determine project root
-require 'pathname'
-project_root= File.expand_path('../..', Pathname.new(__FILE__).realpath)
-
-# Load Bundler
-ENV['BUNDLE_GEMFILE'] ||= (
-  gemfile= File.join(project_root, 'Gemfile')
-  File.exists?(gemfile) && gemfile
-)
-require 'rubygems'
-require 'bundler/setup' if ENV['BUNDLE_GEMFILE']
-
-# Start CLI
-require 'corvid/cli/plugin'
-Corvid::CLI::Plugin.start project_root:        project_root,
-                          plugin_require_path: '<%= require_path %>',
-                          plugin_class_name:   '<%= full_class_name %>'
-
--- corvid-features/corvid.rb	2012-09-07 16:23:54.141993437 +1000
+++ corvid-features/corvid.rb	2012-09-07 16:23:57.122016195 +1000
@@ -13,18 +13,5 @@
   copy_executable 'bin/yard'
   copy_executable 'bin/yardoc'
   copy_executable 'bin/yri'
-  update 2
+  empty_directory 'lib'
 }
-
-update{|ver|
-  case ver
-  when 2
-    template2       'lib/%project_name%/version.rb.tt'
-    template2       '%project_name%.gemspec.tt'
-  end
-}
-
-# Template vars
-def author_name_comment;  author_name  ? '' : '#' end
-def author_email_comment; author_email ? '' : '#' end
-
--- corvid-features/plugin.rb	2012-09-06 07:57:58.255746233 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,8 +0,0 @@
-requirements 'corvid:corvid'
-
-install {
-  empty_directory 'resources'
-  empty_directory 'resources/latest'
-  copy_file       'resources/.gitignore'
-  install_feature 'corvid', 'test_spec', run_bundle_at_exit: true, say_if_installed: false
-}
--- corvid-features/test_spec.rb	2012-09-06 07:57:58.255746233 +1000
+++ corvid-features/test_spec.rb	2012-09-07 16:23:57.122016195 +1000
@@ -1,5 +1,3 @@
-requirements 'corvid:corvid'
-
 install {
   copy_file               '.rspec'
   copy_executable         'bin/guard'
--- corvid-features/test_unit.rb	2012-09-06 07:57:58.255746233 +1000
+++ corvid-features/test_unit.rb	2012-09-07 16:23:57.122016195 +1000
@@ -1,5 +1,3 @@
-requirements 'corvid:corvid'
-
 install {
   copy_executable         'bin/guard'
   copy_file_unless_exists 'test/bootstrap/all.rb'
--- lib/%project_name%/%feature_name%_feature.rb.tt	2012-09-06 11:01:04.906245908 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,19 +0,0 @@
-require 'corvid/feature'
-
-module <%= project_module %>
-  class <%= class_name %> < Corvid::Feature
-
-    since_ver <%= since_ver %>
-
-    managed_install_task? true
-
-    corvid_tasks {
-      # require '<%= project_name %>/example/corvid_task'
-    }
-
-    rake_tasks {
-      # require '<%= project_name %>/example/task.rake'
-    }
-
-  end
-end
--- lib/%project_name%/%plugin_name%_plugin.rb.tt	2012-09-06 07:57:58.255746233 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,44 +0,0 @@
-require 'corvid/plugin'
-#require 'corvid/extension'
-
-module <%= project_module %>
-  class <%= class_name %> < Corvid::Plugin
-    # Plugins can be extensions but are not by default.
-    #include Corvid::Extension
-
-    # The name of the plugin. Must conform to format enforced by {Corvid::NamingPolicy#validate_plugin_name!}.
-    #
-    # @return [String] The plugin name.
-    name '<%= plugin_name %>'
-
-    # The path for Ruby to require in order to load this plugin.
-    #
-    # @return [String] The path to require, usually relative to your lib dir.
-    require_path '<%= require_path %>'
-
-    # The path to the directory containing the plugin's resources.
-    #
-    # @return [String] An absolute path.
-    resources_path File.expand_path('../../../resources', __FILE__)
-
-    # A manifest of all features provided by the plugin.
-    #
-    # @return [Hash<String,Array<String>>] A hash with keys being feature names, and the values being a 2-element
-    #   array of the feature's require-path, and class name, respectively.
-    feature_manifest ({
-    })
-
-    # Callback that is run after the plugin is installed.
-    #
-    # Generator actions are available and can be invoked as if the callback function were a generator method.
-    after_installed {
-      add_dependency_to_gemfile '<%= project_name %>'
-    }
-
-    # A list of features to install automatically when the plugin itself is installed.
-    #
-    # @return [Array<String>] An array of feature names. Do not include the plugin prefix.
-    auto_install_features %w[]
-
-  end
-end
--- lib/%project_name%/version.rb.tt	2012-09-06 07:57:58.255746233 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,5 +0,0 @@
-module <%= project_module %>
-  # The current version of <%= project_module %>.
-  VERSION = "0.0.1".freeze
-end
-
--- resources/.gitignore	2012-09-06 07:57:58.255746233 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1 +0,0 @@
-latest
--- resources/latest/corvid-features/%feature_name%.rb.tt	2012-09-06 11:03:11.953620299 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,23 +0,0 @@
-# Requirements that must be satisfied before the feature can be installed.
-# @return [nil, String, Hash<String,Fixnum|Range|Array<Fixnum>>, Array] Requirements that can be provided to
-#   {RequirementValidator}.
-# @see RequirementValidator#add
-#
-#requirements
-
-install {
-  # empty_directory 'lib'
-  # copy_file       'hello.txt'
-  # copy_executable 'bin/hello'
-
-  # You can also call the update function below to keep things DRY.
-  # update 3
-}
-
-# update {|ver|
-#   case ver
-#   when 2
-#     update stuff here
-#     empty_directory 'lib'
-#   end
-# }
--- test/spec/%plugin_name%_plugin_spec.rb.tt	2012-09-06 07:57:58.255746233 +1000
+++ /dev/null	2012-09-03 14:23:42.005077313 +1000
@@ -1,12 +0,0 @@
-# encoding: utf-8
-require_relative '../bootstrap/spec'
-require 'corvid/test/resource_patch_tests'
-require '<%= require_path %>'
-
-describe <%= full_class_name %> do
-  include Corvid::ResourcePatchTests
-
-  include_resource_patch_tests
-
-  include_feature_update_install_tests
-end
